packages/pages/src/vite-plugin/build/closeBundle/functionMetadata.ts

```diff
@@ -0,0 +1,63 @@
+import fs from "fs";
+import path from "path";
+import esbuild from "esbuild";
+import { importFromString } from "module-from-string";
+import glob from "glob";
+
+const FUNCTIONS_PATH = path.resolve("functions");
+const FUNCTION_METADATA_PATH = path.join(
+  FUNCTIONS_PATH,
+  "functionMetadata.json"
+);
+const TEMP_DIR = ".temp";
+
+/** Metadata for a serverless function. */
+type FunctionMetadata = {
+  /** Name of the function. */
+  entrypoint: string;
+};
+
+/**
+ * Returns a mapping of file path (relative to the repo root) to the metadata
+ * for the function that is the default export of that file.
+ */
+const getFunctionMetadataMap = async (): Promise<
+  Record<string, FunctionMetadata>
+> => {
+  const filepaths = glob
+    .sync(path.join(FUNCTIONS_PATH, "**/*.{tsx,jsx,js,ts}"), { nodir: true })
+    .map((f) => path.resolve(f));
+  const functionMetadataArray: [string, FunctionMetadata][] = await Promise.all(
+    filepaths.map(async (filepath) => {
+      const buildResult = await esbuild.build({
+        entryPoints: [filepath],
+        outdir: TEMP_DIR,
```

Comment:
Do we delete `TEMP_DIR` once we're done?

Reply:
The `TEMP_DIR` isn't actually created and nothing is written to it because we pass in `write: false`